File: backend-Nijan/src/main/java/com/ecommerce/backendNijan/service/ProductService.java
package com.ecommerce.backendNijan.service;

import com.ecommerce.backendNijan.entity.ProductEntity;
import com.ecommerce.backendNijan.model.FilterProduct;
import com.ecommerce.backendNijan.model.IProduct;
import com.ecommerce.backendNijan.model.ProductDto;
import com.ecommerce.backendNijan.response.PageResponse;
import com.ecommerce.backendNijan.response.Product;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.io.IOException;
import java.util.List;
import java.util.Optional;

public interface ProductService {

    /**
     * Get all product.
     *
     * @return list of entity product.
     */
    public List<Product> getAll();

    /**
     * Get all product.
     *
     * @param searchValue searchValue
     * @return list of entity product.
     */
    public PageResponse<ProductDto> getAllBySearchValue(FilterProduct filterProduct, String searchValue);

    /**
     * Get all product.
     *
     * @param filterProduct filterProduct
     * @return list of entity product.
     */
    public PageResponse<ProductDto> getAll(FilterProduct filterProduct);

    /**
     * Get all product.
     *
     * @param categoryId categoryId
     * @param limitItem  limitItem
     * @return list of entity product.
     */
    public List<IProduct> getAll(Long categoryId, Integer limitItem);

    /**
     * Get product item.
     *
     * @return entity product.
     */
    public Product getProductByKey(Long categoryId, Long productId);

    /**
     * Get product item.
     *
     * @return entity product.
     */
    public Optional<ProductEntity> getProductById(Long categoryId, Long productId);

    /**
     * Insert.
     */
    void insert(ProductDto productDto) throws IOException;

    /**
     * Insert.
     */
    void update(ProductDto productDto);

    /**
     * Delete
     * @param productId
     * @return
     */
    boolean delete(Long productId);
}

File: vite-nijan/src/component/modules/ProductService.js
import http from "./http-common";

class ProductService {

    insert(product) {
      return http.post("http://localhost:8888/api/product/insert", product, {
        headers: {
          "Content-Type": "multipart/form-data; charset=utf-8"
        },
      });
    }

    update(productId, product) {
      console.log(product);
      return http.post("http://localhost:8888/api/product/update", product, {
        headers: {
          "Content-Type": "multipart/form-data; charset=utf-8"
        },
      });
    }

    getAllBySearchValue(searchValue) {
      return http.get('http://localhost:8888/api/product/all?search_value=' + searchValue);
    }

    getProductById(categoryId, productId) {
      return http.get('http://localhost:8888/api/product?category_id=' + categoryId + "&product_id=" + productId);
    }

    getAllPageable(page) {
      return http.get('http://localhost:8888/api/product/init/pageable' + "?page_no=" + page.pageNo + "&page_size=" + page.pageSize
      + "&sort_direction=" + page.sortDirection + "&sort_by=" + page.sortBy + "&search_value=" + page.searchValue)
    }

    getAllFilter(page) {
      return http.get('http://localhost:8888/api/product/init/filter' + "?page_no=" + page.pageNo + "&page_size=" + page.pageSize
      + "&sort_direction=" + page.sortDirection + "&sort_by=" + page.sortBy + "&category_id=" + page.categoryId)
    }

    deleteProduct(id) {
      return http.delete("http://localhost:8888/api/product/delete?product_id=" + id);
    }
}

export default new ProductService();
